src/main/thrift/parquet.thrift

```diff
@@ -511,6 +511,51 @@ struct ColumnMetaData {
    * This information can be used to determine if all data pages are
    * dictionary encoded for example **/
   13: optional list<PageEncodingStats> encoding_stats;
+
+  /** Byte offset from beginning of file to bloom filter data**/
+  14: optional i64 bloom_filter_offset;
+}
+
+/**
+  * Definition of bloom filter algorithm.
+  */
+enum BloomFilterAlgorithm {
+  /** Block based bloom filter. 
+   * The bloom filter bitset is separated into tiny bucket as tiny bloom 
+   * filter, the high 32 bits hash value is used to select bucket, and 
+   * lower 32 bits hash values are used to set bits in tiny bloom filter.
+   * See “Cache-, Hash- and Space-Efficient Bloom Filters”. Specifically, 
+   * the tiny bloom filter size is 32 bytes and the algorithm set 8 bits
+   * in each bucket.
+   *
+   * In order to set bits in bucket, the algorithm need 8 SALT values 
+   * (0x47b6137bU, 0x44974d91U, 0x8824ad5bU, 0xa2b7289dU, 0x705495c7U, 
+   * 0x2df1424bU, 0x9efc4947U, 0x5c6bfb31U) to calculate index with formular:
+   *                  index[i] = (hash * SALT[i]) >> 27 
+   **/
+    BLOCK = 0;
+}
+
+/** 
+ * Definition for hash function used to compute hash of column value.
+ * Note that the hash function take plain encoding of column values as input.
+ */
+enum BloomFilterHash {
+  /** Murmur3 has 32 bits and 128 bits hash, we use lower 64 bits from 
+   * Murmur3 128 bits function murmur3hash_x64_128  
+   **/
+    MURMUR3 = 0;
+}
+
+struct BloomFilterHeader {
+  /** The size of bitset in bytes, must be a power of 2 and larger than 32**/
+  1: required i32 numBytes;
+
+  /** The algorithm for setting bits. **/
+  2: required BloomFilterAlgorithm bloomFilterAlgorithm;
```

Comment:
I think the names (but not the type names) of these fields can be simplified by removing "bloomFilter" from the front.